vault/core.go

```diff
@@ -1234,10 +1246,10 @@ func (c *Core) unsealPart(ctx context.Context, seal Seal, key []byte, useRecover
 	} else {
 		masterKey = recoveredKey
 	}
-	newRecoveryKey := masterKey
 
-	// If we have a migration seal, now's the time!
-	if c.migrationSeal != nil {
+	if c.migrationInfo != nil {
+		c.migrationInfo.newRecoveryKey = make([]byte, len(masterKey))
+		copy(c.migrationInfo.newRecoveryKey, masterKey)
```

Comment:
Why copy instead of simple assignment?

Reply:
I had forgotten why I had done this way and resorted for the assignment, only to realize the reason after it broke things. Since the assigned values are reference types, if they get modified after the fact that it is set in `migrationInfo`, it affects the migration process. To be on the safer side, I resorted to copies.

Reply:
Ah, it's because we later pass masterKey to AESGCMBarrier.Unseal, which does a memzero.  I'd like to figure out what our posture on that is: in the barrier code we seem concerned about not keeping master keys around in memory, but we do so plenty elsewhere.  Feels like we should be more consistent about it, and document that behaviour in functions that have that side effect.  For now I agree with your approach.

Reply:
In more recent times, I've seen us not being very concerned about keeping keys in memory. Mainly because we can not really wipe it out of memory. It is basically covered by our threat model. Regarding passing the keys in this specific case, I had discussed with BrianK too since I had a similar thought as yours and we both landed that its okay to do it this way.

Reply:
I don't think it's wrong to do it this way, but like I say I'd rather we were consistent.  If we don't need to memzero master keys, let's get rid of the instances we still have so we can stop having to worry about these side effects.  (Not in this PR, I'll open another for that.)